app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentPresenter.kt

```diff
@@ -120,15 +134,52 @@ class FeatureFlagsFragmentPresenter @Inject constructor(
   ) {
     binding.viewModel = model
 
+    binding.resetButton.setOnClickListener {
+      handleResetFeatureFlag(model)
+    }
+    if (resetFlags.containsKey(model.featureFlagId)) {
+      model.isFlagOverridden.set(true)
+      model.isResetButtonActive.set(false)
+    }
+
     if (featureFlagStates.containsKey(model.featureFlagId)) {
       model.isChecked.set(featureFlagStates[model.featureFlagId])
     }
     model.onFeatureFlagToggleCallback = { id, value ->
-      if (model.currentValue == value) {
+      if (model.currentValue == value && !resetFlags.containsKey(id)) {
```

Comment:
Nit: a bit more idiomatic in Kotlin and reads more clearly as “id is not in resetFlags.”
```suggestion
      if (model.currentValue == value && id !in resetFlags) {
```

Reply:
updated